src/rust/demikernel/bindings.rs

```diff
@@ -512,10 +512,28 @@ pub extern "C" fn demi_getsockopt(
 // Standalone Functions
 //==============================================================================
 
-/// Converts a [sockaddr] into a [Ipv4Endpoint].
+/// Converts a [sockaddr] into a [SocketAddrV4].
 fn sockaddr_to_ipv4endpoint(saddr: *const sockaddr) -> Result<SocketAddrV4, Fail> {
-    // TODO: Review why we need byte ordering conversion here.
     let sin: libc::sockaddr_in = unsafe { *mem::transmute::<*const sockaddr, *const libc::sockaddr_in>(saddr) };
-    let addr: Ipv4Addr = { Ipv4Addr::from(u32::from_be_bytes(sin.sin_addr.s_addr.to_le_bytes())) };
-    Ok(SocketAddrV4::new(addr, sin.sin_port))
+    let addr: Ipv4Addr = Ipv4Addr::from(sin.sin_addr.s_addr.to_be());
+    let port: u16 = sin.sin_port.to_be();
```

Comment:
Do you need these "to_be()" things?  I find it jarring to see them there, as they shouldn't do anything.  In any sort of socket address, the address and port numbers are always supposed to be in Network Byte Order, which was decreed by the gods, or at least Jon Postel*, to be big-endian.

* I don't know for sure it was Jon, but he seems like a likely candidate.

Reply:
And if you do need them for this code to work correctly, then they're wrong.

Reply:
> And if you do need them for this code to work correctly, then they're wrong. Since these values are big-endian by definition, calling "to_be()" on them actually turns them little-endian. Which is host order on x86. So this works until we try to run it on a big-endian machine.

We do need them. I wrote the unit test to precisely illustrate this.

Any of the following alternatives is buggy:
- Drop calls `to_be()`
- Replace `to_be()` for `to_le()`
- Replace `to_be()` for `to_ne()`

Note that before this PR we were not performing a conversion on the port number, and we were operating on bad port numbers. You can check this on unmodified code by turning on rust logging.

Reply:
I understand all that.  This is a subtle, but important point.

We do need a byte-swap (on x86) for this to work correctly.  But we should do it in a way that is both portable (to big-endian architectures) and makes sense to the reader.

Here's the issue:
- A sockaddr (the C struct used by the Berkeley "sockets" interface) is **always** in network byte order.
> - And network-byte order is **always** big-endian.
- Rust's SockAddrV4 constructor takes an Ipv4Addr (ip address) and an u16 (port number).
> - The u16 is expected to be in host order.
> - The constructor for Rust's Ipv4Addr struct that we're using takes an u32, which is expected to be in host order.

So we're taking something that is always network byte order (big-endian) and converting it to host order so we can create the Rust structure.  A human reading this code would expect whatever conversion function we call to reflect this in its name.  So something named network_to_host_endian() is what we want here.

Reply:
As we discussed I've updated the code to rely on `from_be()` instead of `to_be()`.